README.md

```diff
@@ -635,6 +635,29 @@ function hello($name, LoopInterface $loop)
 hello('Tester', $loop);
 ```
 
+If you want to create a timeout rather than limiting the number of executions,
+you can track timers in an array and pass it by reference. This is useful when
+waiting for another method or service to be ready.
+
+```php
+$timers = [];
+$cancelTimers = static function (array &$timers) {
+  array_map('\React\EventLoop\Loop::cancelTimer', $timers);
+  $timers = [];
+};
+$timers[] = Loop::addPeriodicTimer(5, function () use (&$timers, $cancelTimers) {
+    if (someExternalService()) {
+      $cancelTimers($timers);
+      echo 'someExternalService is ready' . PHP_EOL;
+    }
+    echo 'Waiting for someExternalService' . PHP_EOL;
+});
+$timers[] = Loop::addTimer(60*45, function () use (&$timers, $cancelTimers) {
+    $cancelTimers($timers);
+    echo 'Timed out waiting for someExternalService' . PHP_EOL;
+});
+```
```

Comment:
There is no need to to also pass in `$timers` by reference if you use the function approach, you can do that directly on the function.

```php
$timers = [];
$cancelTimers = static function () use (&$timers) {
  array_map('\React\EventLoop\Loop::cancelTimer', $timers);
  $timers = [];
};
$timers[] = Loop::addPeriodicTimer(5, function () use ($cancelTimers) {
    if (someExternalService()) {
      $cancelTimers();
      echo 'someExternalService is ready' . PHP_EOL;
    }
    echo 'Waiting for someExternalService' . PHP_EOL;
});
$timers[] = Loop::addTimer(60*45, function () use ($cancelTimers) {
    $cancelTimers();
    echo 'Timed out waiting for someExternalService' . PHP_EOL;
});
```

Reply:
Good catch! I cleaned up the examples.